src/events.cpp

```diff
@@ -736,6 +753,35 @@ void peek_for_resize()
 	}
 }
 
+void call_in_main_thread(const std::function<void(void)>& f)
+{
+	if (boost::this_thread::get_id() == main_thread) {
+		// nothing special to do if called from the main thread.
+		f();
+		return;
+	}
+
+
+	invoked_funktion_data fdata = { false, f };
+
+	SDL_Event sdl_event;
+	SDL_UserEvent sdl_userevent;
+
+	sdl_userevent.type = INVOKE_FUNCTION_EVENT;
```

Comment:
he problem here is that this code really depends on the function beeing called in the main thread, becasue otherwide the application won't continue. the SDL_Timer events, are part of the usual event queue, and the code in events.cpp::poll has code to skip events under some unkown conditions. So  so make sure these get always executed this was made to be handled called earlier than the 'usual event queue'